src/include/ipc/probe.h

```diff
@@ -0,0 +1,134 @@
+
+#define PROBE_SHIFT_FMT_TYPE		31
+#define PROBE_SHIFT_STANDARD_TYPE	27
+#define PROBE_SHIFT_AUDIO_FMT		23
+#define PROBE_SHIFT_SAMPLE_RATE		19
+#define PROBE_SHIFT_NB_CHANNELS		14
+#define PROBE_SHIFT_SAMPLE_SIZE		12
+#define PROBE_SHIFT_CONTAINER_SIZE	10
+#define PROBE_SHIFT_SAMPLE_FMT		9
+#define PROBE_SHIFT_SAMPLE_END		8
+#define PROBE_SHIFT_INTERLEAVING_ST	7
+
+#define PROBE_MASK_FMT_TYPE		MASK(31, 31)
+#define PROBE_MASK_STANDARD_TYPE	MASK(30, 27)
+#define PROBE_MASK_AUDIO_FMT		MASK(26, 23)
+#define PROBE_MASK_SAMPLE_RATE		MASK(22, 19)
+#define PROBE_MASK_NB_CHANNELS		MASK(18, 14)
+#define PROBE_MASK_SAMPLE_SIZE		MASK(13, 12)
+#define PROBE_MASK_CONTAINER_SIZE	MASK(11, 10)
+#define PROBE_MASK_SAMPLE_FMT		MASK(9, 9)
+#define PROBE_MASK_SAMPLE_END		MASK(8, 8)
+#define PROBE_MASK_INTERLEAVING_ST	MASK(7, 7)
+
+/**
+ * Header for data packets sent via compressed PCM from extraction probes
+ */
+struct probe_data_packet {
+	uint32_t sync_word;		/**< PROBE_EXTRACT_SYNC_WORD */
+	uint32_t buffer_id;		/**< Buffer ID from which data was extracted */
+	uint32_t format;		/**< Encoded data format */
+	uint32_t timestamp_low;		/**< Low 32 bits of timestamp in us */
+	uint32_t timestamp_high;	/**< High 32 bits of timestamp in us */
+	uint32_t checksum;		/**< CRC32 of header and payload */
+	uint32_t data_size_bytes;	/**< Size of following audio data */
+	uint32_t data[];		/**< Audio data extracted from buffer */
+} __attribute__((packed));
+
+/**
+ * Description of probe dma
+ */
+struct probe_dma {
+	uint32_t stream_tag;		/**< Stream tag associated with this DMA */
+	uint32_t dma_buffer_size;	/**< Size of buffer associated with this DMA */
+} __attribute__((packed));
+
+/**
+ * Description of probe point
+ */
+struct probe_point {
+	uint32_t buffer_id;	/**< ID of buffer to which probe is attached */
+	uint32_t purpose;	/**< PROBE_PURPOSE_EXTRACTION or PROBE_PURPOSE_INJECTION */
+	uint32_t stream_tag;	/**< Stream tag of DMA via which data will be provided for injection */
+} __attribute__((packed));
+
+/**
+ * \brief DMA ADD for probes.
+ *
+ * Used as payload for IPCs: SOF_IPC_PROBE_INIT, SOF_IPC_PROBE_DMA_ADD.
+ */
+struct sof_ipc_probe_dma_add_params {
+	struct sof_ipc_cmd_hdr hdr;	/**< Header */
+	struct probe_dma probe_dma[];	/**< Array of DMAs to be added */
+} __attribute__((packed));
+
+/**
+ * \brief Reply to INFO functions.
+ *
+ * Used as payload for IPCs: SOF_IPC_PROBE_DMA_INFO, SOF_IPC_PROBE_POINT_INFO.
+ */
+struct sof_ipc_probe_info_params {
+	struct sof_ipc_reply rhdr;			/**< Header */
+	union {
+		struct probe_dma probe_dma[0];		/**< DMA info */
+		struct probe_point probe_point[0];	/**< Probe Point info */
+	};
+} __attribute__((packed));
+
+/**
+ * \brief Probe DMA remove.
+ *
+ * Used as payload for IPC: SOF_IPC_PROBE_DMA_REMOVE
+ */
+struct sof_ipc_probe_dma_remove_params {
+	struct sof_ipc_cmd_hdr hdr;	/**< Header */
+	uint32_t stream_tag[];		/**< Array of stream tags associated with DMAs to remove */
```

Comment:
how do you know how many elements are in the array?

Reply:
Same as above, IPC header.size

Reply:
add the info in the comments to avoid any ambiguity.

Reply:
done